web3.js/src/ed25519-program.ts

```diff
@@ -87,12 +97,12 @@ export class Ed25519Program {
         numSignatures,
         padding: 0,
         signatureOffset,
-        signatureInstructionIndex: instructionIndex,
+        signatureInstructionIndex: instructionIndex ?? 0,
         publicKeyOffset,
-        publicKeyInstructionIndex: instructionIndex,
+        publicKeyInstructionIndex: instructionIndex ?? 0,
         messageDataOffset,
         messageDataSize: message.length,
-        messageInstructionIndex: instructionIndex,
+        messageInstructionIndex: instructionIndex ?? 0,
```

Comment:
I chose to default these to zero, because – ostensibly – encoding a struct with `undefined` u8s results in `0` materializing when you _decode_ it. I thought it better to be explicit about what's going _in_ to this buffer in light of knowing what's going to come _out_ of it.

Reply:
Thoughts, @seanyoung?

Reply:
Sounds reasonable!

Reply:
If the value of the instruction index is `u16::MAX` (i.e. 0xffff) then it will default to the *current* instruction. This should make the devx slightly better. See https://github.com/solana-labs/solana/commit/2400e86d136f5c151eb447bd7fbda6fed4068f49

Reply:
I made this change, @seanyoung. Thanks for the tip!